runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -553,14 +555,23 @@ public TailsResponse getAllTails() {
     }
 
     /**
-     * Get log address space, which includes:
+     * Get log address space of all streams, which includes:
      *     1. Addresses belonging to each stream.
      *     2. Log Tail.
      * @return
      */
     public StreamsAddressResponse getLogAddressSpace() {
-        return layoutHelper(
-                e -> Utils.getLogAddressSpace(e.getLayout(), runtime));
+        return getLogAddressSpace(StreamsAddressRequest.TOTAL);
+    }
+
+    /**
+     * Gets address space of streams in interest
+     *
+     * @param request wrapper of streams in interest.
+     * @return Get log address space, which includes: 1. Addresses belonging to each stream. 2. Log Tail.
+     */
+    public StreamsAddressResponse getLogAddressSpace(StreamsAddressRequest request) {
```

Comment:
This new API is not used other than the wrapped API above. I think we don't need this.

Reply:
It is true that the function is only used in another wrapper function but this api is more flexible.

Reply:
I reverted back to using only one API.